src/puppetlabs/trapperkeeper/services/webserver/jetty9_config.clj

```diff
@@ -52,23 +52,22 @@
 ;;; their implicit defaults reflect the security issue.  The latter is far more
 ;;; risky for our downstream apps, thus it was decided that it makes sense to
 ;;; keep these overrides.
-;;;
-;;; Also note that w/rt the default list of acceptable ciphers, we're deliberately
-;;; excluding all diffie-helman ciphers due to some old JDK bugs:
-;;;
-;;; http://bugs.java.com/bugdatabase/view_bug.do?bug_id=8014618
-;;; https://github.com/puppetlabs/puppetdb/commit/03e020dc85b83d6c83c9992ca6bd14f57e8fc91a
-;;;
-;;; These have been fixed in recent versions of the JDK, and it would be nice
-;;; to be able to add the DH ciphers back in at some point, but we can't do that
-;;; until we're certain that our minimum supported JDK versions for all of our
-;;; supported distros will contain the relevant fixes.
-;;;
 (def acceptable-ciphers
-  ["TLS_RSA_WITH_AES_256_CBC_SHA256"
+  ["TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA256"
+   "TLS_ECDHE_RSA_WITH_AES_256_CBC_SHA"
+   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA256"
+   "TLS_ECDHE_RSA_WITH_AES_128_CBC_SHA"
```

Comment:
First off, my knowledge of the different ECC curves is a bit limited, so take this with a grain of sale.

I haven't been able to track down which curves can specifically be used with ECDHE. There's documentation for ECDSA and ECGOST (https://www.bouncycastle.org/wiki/pages/viewpage.action?pageId=362269) which is a starting point, but it includes a number of curves that (in my uneducated guess) seem to be only used with ECDSA.

The Bouncy Castle ECDHE key exchange implementation (https://github.com/bcgit/bc-java/blob/master/core/src/main/java/org/bouncycastle/crypto/tls/TlsECDHEKeyExchange.java) references a list of curve types (https://github.com/bcgit/bc-java/blob/master/core/src/main/java/org/bouncycastle/crypto/tls/TlsECCUtils.java#L35-L39) that look like they'll be used when selecting a curve.

As I was writing this I saw some functions that seemed like they might explain things a bit, I'm going to dive into those deeper before I expand on this comment.